massa-ledger-exports/src/key.rs

```diff
@@ -88,12 +90,27 @@ impl Serializer<Vec<u8>> for KeySerializer {
     /// KeySerializer::new().serialize(&key, &mut serialized).unwrap();
     /// ```
     fn serialize(&self, value: &Vec<u8>, buffer: &mut Vec<u8>) -> Result<(), SerializeError> {
-        let limit = ADDRESS_SIZE_BYTES + 1;
-        buffer.extend(&value[..limit]);
-        if value[limit - 1] == DATASTORE_IDENT {
-            if value.len() > limit {
-                self.vec_u8_serializer
-                    .serialize(&value[limit..].to_vec(), buffer)?;
+        let result_addr = match self
+            .address_deserializer
+            .deserialize::<massa_serialization::DeserializeError>(value)
+        {
+            Ok((rest, addr)) => Ok((rest, addr)),
+            Err(_) => Err(SerializeError::GeneralError(String::from(
+                "The key does not contain a valid address",
+            ))),
+        };
+        if result_addr.is_err() {
+            return Err(SerializeError::GeneralError(String::from(
+                "The key does not contain a valid address",
+            )));
+        }
+        let (rest, addr): (&[u8], Address) = result_addr.unwrap();
+
+        buffer.extend(addr.to_prefixed_bytes());
+
+        if rest[0] == DATASTORE_IDENT {
+            if !rest.is_empty() {
+                self.vec_u8_serializer.serialize(&rest.to_vec(), buffer)?;
```

Comment:
Double check that the byte-vec is an encoding of a u8, as opposed to a serialized varint that happens to only need 1 byte. Might be better to do a deserialization on `rest` using something like the `nom::verify` thing?